common/src/main/java/com/unrealdinnerbone/jamd/util/TelerportUtils.java

```diff
@@ -60,15 +60,18 @@ private static Optional<BlockPos> findPortalLocation(Level worldTo, BlockPos fro
     }
 
     private static boolean forLocationAround(Level levelTo, BlockPos.MutableBlockPos blockPos, int fromX, int fromZ, int y) {
-        for (int x = fromX - 6; x < fromX + 6; x++) {
-            for (int z = fromZ - 6; z < fromZ + 6; z++) {
+        boolean locationGood = false;
+        for (int x = fromX - 4; x < fromX + 4; x++) {
+            if(locationGood) { break; }
```

Comment:
Please format this like the rest of the code

Reply:
![image](https://github.com/nanite/JAMD/assets/19319115/057ddde1-647a-418c-ade7-711a7528def4)

I believe the formatting is only different because you're using eclipse and I'm using ideaCommunity. Sorry about that. 

My explanation for changing the values is null now, because you changed getBlock to getBlockEntity and findPortalLocation is now able to find the portal correctly. Before the method would always fail at the findFirst sub-method and go to the else block. I was tweaking a bunch of little values to fix spawning on top of buildings, but now that the method actually finds the entity block correctly, the locationAround method never gets called, and the player gets spawned correctly, even with a only 2 high surface above them.